composer/utils/object_store/s3_object_store.py

```diff
@@ -122,19 +139,20 @@ def get_object_size(self, object_name: str) -> int:
             _ensure_not_found_errors_are_wrapped(self.get_uri(object_name), e)
         return obj['ContentLength']
 
-    def upload_object(
-        self,
-        object_name: str,
-        filename: Union[str, pathlib.Path],
-        callback: Optional[Callable[[int, int], None]] = None,
-    ):
+    def upload_object(self,
+                      object_name: str,
+                      filename: Union[str, pathlib.Path],
+                      callback: Optional[Callable[[int, int], None]] = None,
+                      canned_acl: Optional[S3CannedACL] = None):
```

Comment:
```suggestion
                      canned_acl: Optional[Union[str, S3CannedACL]] = None):
```
I would really prefer an option to specify a string here makes it way easier if you know the string name by heart